micrometer/src/main/scala/zio/metrics/connectors/micrometer/MicrometerMetricListener.scala

```diff
@@ -0,0 +1,113 @@
+package zio.metrics.connectors.micrometer
+
+import java.lang.{Double => JDouble}
+import java.time.Instant
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicLong
+
+import scala.jdk.CollectionConverters.IterableHasAsJava
+
+import zio.{Unsafe, ZIO}
+import zio.metrics.{MetricKey, MetricKeyType, MetricListener}
+import zio.metrics.MetricKeyType.{Counter, Frequency, Gauge}
+
+import io.micrometer.core.instrument.{DistributionSummary, MeterRegistry, Tag}
+
+private[micrometer] class MicrometerMetricListener(meterRegistry: MeterRegistry) extends MetricListener {
+  private lazy val gauges: ConcurrentHashMap[MetricKey[Gauge], AtomicDouble] =
+    new ConcurrentHashMap[MetricKey[Gauge], AtomicDouble]()
+
+  override def updateHistogram(
+    key: MetricKey[MetricKeyType.Histogram],
+    value: Double,
+  )(implicit unsafe: Unsafe,
+  ): Unit =
+    DistributionSummary
+      .builder(key.name)
```

Comment:
It seems that the name of the metric should conform to the [convention](https://micrometer.io/docs/concepts#_naming_meters). 
See PrometheusEncoder#encodeMetric

Reply:
I thought about it but decided to let user set basic name of metric key that would be thrown further. After the metric is processed by the vendor-specific Micrometer backend, the metric key name is automatically formatted according to the conventions of that backend. For example, when using PrometheusMeterRegistry, counters have "_total" appended to the metric name, and duration metrics have "_seconds" appended, etc. 

Please let me know if you have any further concerns or suggestions regarding this approach.